engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/ScriptAssertContext.java

```diff
@@ -27,25 +29,26 @@
 	private static final Log log = LoggerFactory.make();
 
 	private final String script;
-	private final ScriptEvaluator scriptEvaluator;
+	private final String languageName;
 
 	public ScriptAssertContext(String languageName, String script) {
 		this.script = script;
-		this.scriptEvaluator = getScriptEvaluator( languageName );
+		this.languageName = languageName;
 	}
 
-	public boolean evaluateScriptAssertExpression(Object object, String alias) {
+	public boolean evaluateScriptAssertExpression(Object object, String alias, ConstraintValidatorContext context) {
 		Map<String, Object> bindings = newHashMap();
 		bindings.put( alias, object );
 
-		return evaluateScriptAssertExpression( bindings );
+		return evaluateScriptAssertExpression( bindings, context );
 	}
 
-	public boolean evaluateScriptAssertExpression(Map<String, Object> bindings) {
+	public boolean evaluateScriptAssertExpression(Map<String, Object> bindings, ConstraintValidatorContext context) {
```

Comment:
Right ... But it's either here or via the constructor right? And in case of constructor we'd need to get it in the `initailize` method:

```java
public void initialize(ParameterScriptAssert constraintAnnotation) {
   // get the Context somehow
}
```

Or did you have something different on your mind?

Reply:
No it was more about building what's needed in `ParameterScriptAssertValidator` and pass the `ScriptEvaluator` directly.

About the `initialize()`, that's what I posted in another message, it's too bad we don't have a `Context` object here, it would have been useful.